tests/test_gpu_common.py

```diff
@@ -269,6 +270,7 @@ def test_tasking_lock_placement(self):
         assert str(sections[1].body[0].body[0].body[0].body[0]) ==\
             'while(lock0[t1] == 0);'
 
+    @skipif('device-openmp')  # TODO: Still unsupported with OpenMP, but soon will be
```

Comment:
Perhaps we need to change the policy here for TODO's. At the time of writing the developer likely has a clear understanding of what needs to be done; but clarity can quickly evaporate as the developer undertakes other tasks. 

One possibility is to link every TODO with a corresponsing issue number. There the developer can add their notes for their future self or whoever picks up the TODO later.

Reply:
OK. But the "future" you're talking about here is more like imminent future (hence the "soon"). Basically all these TODOs point to the AMD issues... will see how to improve it